### PR TITLE
Patch jQuery update fixes since WP 5.6

### LINKED ISSUE #5892: jQuery update since WP 5.6
WP 5.6 ships with an updated jQuery version. We should patch any deprecated methods we call in jQuery and update libraries where possible.

Also fix: https://wordpress.org/support/topic/unable-to-create-new-pod-in-wordpress-5-6

### PR DESCRIPTION
**Pull request by JoryHogeveen**
Fixes: #5892 
Fixes: #5896
Fixes: #5897

## PR Checklist

- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: ui/fields/_db.php [hunk 1]
```diff
@@ -8,7 +8,7 @@
 <input<?php PodsForm::attributes( $attributes, $name, $form_field_type, $options ); ?> />
 <script>
 	jQuery( function ( $ ) {
-		$( 'input#<?php echo esc_js( $attributes['id'] ); ?>' ).change( function () {
+		$( 'input#<?php echo esc_js( $attributes['id'] ); ?>' ).on( 'change', function () {
```
Comment (tr1b0t) [inline ui/fields/_db.php:11]: :no_entry_sign: Barf out, gag me with a spoon! esc_js()!This is for inline javascript, which is against our standards. Use json_encode instead.

<sub>hash: ac1e84b6c5871e71fdb3b2ea351e870e</sub>

File: ui/fields/slug.php [hunk 2]
```diff
@@ -8,7 +8,7 @@
 <input<?php PodsForm::attributes( $attributes, $name, $form_field_type, $options ); ?> />
 <script>
 	jQuery( function ( $ ) {
-		$( 'input#<?php echo esc_js( $attributes['id'] ); ?>' ).change( function () {
+		$( 'input#<?php echo esc_js( $attributes['id'] ); ?>' ).on( 'change', function () {
```
Comment (tr1b0t) [inline ui/fields/slug.php:11]: :no_entry_sign: Barf out, gag me with a spoon! esc_js()!This is for inline javascript, which is against our standards. Use json_encode instead.

<sub>hash: 3d6cb6772c3324ca4ffda81002638f11</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 2 error(s)

File: ui/admin/setup-add.php [hunk 3]
```diff
@@ -423,16 +422,19 @@
...
+		$document.Pods( 'validate' );
+		$document.Pods( 'submit' );
+		$document.Pods( 'wizard' );
+		$document.Pods( 'dependency' );
+		$document.Pods( 'advanced' );
+		$document.Pods( 'confirm' );
+		$document.Pods( 'sluggable' );
 
+		$document.find( '.pods-dependent-toggle[data-name-clean]' ).trigger( 'change' );
 		$( '.pods-admin' ).on( 'render', '.pods-form-ui-field', function ( e ) {
```
Comment (sc0ttkclark) [inline ui/admin/setup-add.php:436]:
```suggestion

		var $admin = $( '.pods-admin' );

		$admin.on( 'render', '.pods-form-ui-field', function ( e ) {
```

File: ui/admin/setup-add.php [hunk 4]
```diff
@@ -423,16 +422,19 @@
...
-			$( this ).find( '.pods-dependent-toggle[data-name-clean]' ).trigger( 'change' );
+			$( '.pods-admin' ).find( '.pods-dependent-toggle[data-name-clean]' ).trigger( 'change' );
```
Comment (sc0ttkclark) [inline ui/admin/setup-add.php:437]:
```suggestion
			$toggles.trigger( 'change' );
```

File: ui/admin/setup-add.php (hunk 3 above)
Comment (JoryHogeveen) [reply ui/admin/setup-add.php:436]: Not sure what this would fix? The `$document` relation (directly on load) is important here

Comment (JoryHogeveen) [reply ui/admin/setup-add.php:436]: Oh wait, it's about the line below that haha, sure looks good to me!

File: classes/PodsInit.php [hunk 5]
```diff
@@ -490,7 +490,7 @@ public function register_assets() {
 		wp_register_script( 'pods-json', PODS_URL . 'ui/js/jquery.json.js', array( 'jquery' ), '2.3' );
 
 		if ( ! wp_script_is( 'jquery-qtip2', 'registered' ) ) {
-			wp_register_script( 'jquery-qtip2', PODS_URL . 'ui/js/jquery.qtip.min.js', array( 'jquery' ), '2.2' );
+			wp_register_script( 'jquery-qtip2', PODS_URL . "ui/js/qtip/jquery.qtip{$suffix_min}.js", array( 'jquery' ), '3.0.3' );
```
Comment (tr1b0t) [inline classes/PodsInit.php:493]: * :warning: In footer ($in_footer) is not set explicitly wp_register_script; It is recommended to load scripts in the footer. Please set this value to `true` to load it in the footer, or explicitly `false` if it should be loaded in the header.
* :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 3b7f0c0d12a678e3134f352578f413dc</sub>

File: classes/PodsInit.php [hunk 6]
```diff
@@ -507,7 +507,7 @@ public function register_assets() {
 			true
 		);
 
-		wp_register_script( 'pods-cleditor', PODS_URL . 'ui/js/jquery.cleditor.min.js', array( 'jquery' ), '1.3.0' );
+		wp_register_script( 'pods-cleditor', PODS_URL . "ui/js/cleditor/jquery.cleditor{$suffix_min}.js", array( 'jquery' ), '1.4.5' );
```
Comment (tr1b0t) [inline classes/PodsInit.php:510]: * :warning: In footer ($in_footer) is not set explicitly wp_register_script; It is recommended to load scripts in the footer. Please set this value to `true` to load it in the footer, or explicitly `false` if it should be loaded in the header.
* :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: b0f8e64f10042d7b1967540893f0459c</sub>

File: includes/general.php [hunk 7]
```diff
@@ -532,7 +532,7 @@ function pods_deprecated( $function, $version, $replacement = null ) {
 function pods_help( $text, $url = null ) {
 
 	if ( ! wp_script_is( 'jquery-qtip2', 'registered' ) ) {
-		wp_register_script( 'jquery-qtip2', PODS_URL . 'ui/js/jquery.qtip.min.js', array( 'jquery' ), '2.2' );
+		wp_register_script( 'jquery-qtip2', PODS_URL . 'ui/js/qtip/jquery.qtip.min.js', array( 'jquery' ), '3.0.3' );
```
Comment (tr1b0t) [inline includes/general.php:535]: * :warning: In footer ($in_footer) is not set explicitly wp_register_script; It is recommended to load scripts in the footer. Please set this value to `true` to load it in the footer, or explicitly `false` if it should be loaded in the header.
* :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 14a540264975d6c6e5c6c69d0c661d7a</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 5 error(s)
:warning: 3 warning(s)

File: classes/PodsInit.php [hunk 8]
```diff
@@ -487,10 +487,10 @@ public function register_assets() {
 		 */
 		do_action( 'pods_before_enqueue_scripts', $suffix_min );
 
-		wp_register_script( 'pods-json', PODS_URL . 'ui/js/jquery.json.js', array( 'jquery' ), '2.3' );
+		wp_register_script( 'pods-json', PODS_URL . 'ui/js/jquery.json.js', array( 'jquery' ), '2.3', true );
```
Comment (tr1b0t) [inline classes/PodsInit.php:490]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: d5ae3a69e1ad1dfc1a7df08185d7347d</sub>

File: classes/PodsInit.php [hunk 9]
```diff
@@ -487,10 +487,10 @@ public function register_assets() {
...
 
 		if ( ! wp_script_is( 'jquery-qtip2', 'registered' ) ) {
-			wp_register_script( 'jquery-qtip2', PODS_URL . 'ui/js/jquery.qtip.min.js', array( 'jquery' ), '2.2' );
+			wp_register_script( 'jquery-qtip2', PODS_URL . "ui/js/qtip/jquery.qtip{$suffix_min}.js", array( 'jquery' ), '3.0.3', true );
```
Comment (tr1b0t) [inline classes/PodsInit.php:493]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 0c3bdcc81a91eaf24a3ff6a074709d0c</sub>

File: classes/PodsInit.php [hunk 10]
```diff
@@ -507,7 +507,7 @@ public function register_assets() {
 			true
 		);
 
-		wp_register_script( 'pods-cleditor', PODS_URL . 'ui/js/jquery.cleditor.min.js', array( 'jquery' ), '1.3.0' );
+		wp_register_script( 'pods-cleditor', PODS_URL . "ui/js/cleditor/jquery.cleditor{$suffix_min}.js", array( 'jquery' ), '1.4.5', true );
```
Comment (tr1b0t) [inline classes/PodsInit.php:510]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 7bb5203408358d75029ad6dd8c47fd8f</sub>

File: classes/PodsInit.php [hunk 11]
```diff
@@ -574,10 +574,10 @@ public function register_assets() {
 					'jquery',
 					'pods-i18n',
 				),
-				'1.0.1',
+				'1.0.8',
 				true
 			);
-			wp_register_script( 'pods-select2', $select2_i18n, array( 'pods-select2-core' ), '1.0.1', true );
+			wp_register_script( 'pods-select2', $select2_i18n, array( 'pods-select2-core' ), '1.0.8', true );
```
Comment (tr1b0t) [inline classes/PodsInit.php:580]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 849a138efeb6568633f00f9c090f9671</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 7 error(s)
:warning: 1 warning(s)

**Comment by JoryHogeveen** [comment]
@sc0ttkclark 
List of JS libs that use deprecated methods:

- jQuery JSON - last update: v2.3 - 2011
- CLEditor - last update: v1.4.5 (this PR) - 2014
- Timepicker - last update: v1.6.3 - 2016
  - jQuery UI Slider Access- last update: v0.3 - 2012
- qTip - last update: v3.0.3 (this PR) - 2016
- selectWoo - last update: v1.0.8 - 2020

**Review by sc0ttkclark — DISMISSED** [review]

File: classes/PodsInit.php [hunk 12]
```diff
@@ -487,10 +487,8 @@ public function register_assets() {
 		 */
 		do_action( 'pods_before_enqueue_scripts', $suffix_min );
 
-		wp_register_script( 'pods-json', PODS_URL . 'ui/js/jquery.json.js', array( 'jquery' ), '2.3' );
-
 		if ( ! wp_script_is( 'jquery-qtip2', 'registered' ) ) {
-			wp_register_script( 'jquery-qtip2', PODS_URL . 'ui/js/jquery.qtip.min.js', array( 'jquery' ), '2.2' );
+			wp_register_script( 'jquery-qtip2', PODS_URL . "ui/js/qtip/jquery.qtip{$suffix_min}.js", array( 'jquery' ), '3.0.3', true );
```
Comment (tr1b0t) [inline classes/PodsInit.php:491]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: d5754925950536221768d0da749a06c5</sub>

File: classes/PodsInit.php [hunk 13]
```diff
@@ -500,14 +498,13 @@ public function register_assets() {
...
 				'pods-i18n',
-				'pods-json',
 				'jquery-qtip2',
 			),
 			PODS_VERSION,
 			true
 		);
 
-		wp_register_script( 'pods-cleditor', PODS_URL . 'ui/js/jquery.cleditor.min.js', array( 'jquery' ), '1.3.0' );
+		wp_register_script( 'pods-cleditor', PODS_URL . "ui/js/cleditor/jquery.cleditor{$suffix_min}.js", array( 'jquery' ), '1.4.5', true );
```
Comment (tr1b0t) [inline classes/PodsInit.php:507]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 73334969fd30e458d79abe2476fdb1f2</sub>

File: classes/PodsInit.php [hunk 14]
```diff
@@ -574,10 +571,10 @@ public function register_assets() {
 					'jquery',
 					'pods-i18n',
 				),
-				'1.0.1',
+				'1.0.8',
 				true
 			);
-			wp_register_script( 'pods-select2', $select2_i18n, array( 'pods-select2-core' ), '1.0.1', true );
+			wp_register_script( 'pods-select2', $select2_i18n, array( 'pods-select2-core' ), '1.0.8', true );
```
Comment (tr1b0t) [inline classes/PodsInit.php:577]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 74b1b1c06020dc630e610d3d21cee18d</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 6 error(s)
:warning: 1 warning(s)

**Review by sc0ttkclark — APPROVED** [review]

**Pull request merged**